applications/klighd-cli/src/popup.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * KIELER - Kiel Integrated Environment for Layout Eclipse RichClient
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ *
+ * Copyright 2021 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ *
+ * This code is provided under the terms of the Eclipse Public License (EPL).
+ */
+
+import "./styles/popup.css";
+
+interface PopupOptions {
+    /** The created popup will not be hidden automatically. */
+    persist: boolean;
+}
+
+type PopupType = "info" | "warn" | "error";
```

Comment:
Unions also lead to errors at compile time, so no advantage there. And they have less runtime overhead (enums are compiled to an IIFE, only const enums are inlined but are generally avoided). In fact, unions seems to be preferred by the JS open source eco-system as most libraries use unions instead of enums.
But I am willing to change it. Your call.

Reply:
I remember seeing you putting "warn" in some string somewhere directly which could cause problems if it is only typed with `string`. If it is is still typed with `PopupType` it is safe and if most libraries do it the same, I am fine with this.

Reply:
To my knowledge, I have used [literal types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#literal-types) with unions or [const enums](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-1-4.html#const-enum-completely-inlined-enums) (NotificationType in Services) for all configuration parameters and not basic strings.
If I have used a simple string somewhere, let me know when you find such occurrence! That would be a major downside and should be fixed. Otherwise, I would consider this conversation as resolved.

Sidenote:
I wrote above that const enums are generally avoided and now I present an example where I use a const enum. The TS docs actually do not state any downside and I find the inlining aspect of const enums vs. runtime overhead for simple config options quite useful.
However, TypeScript code is not always compiled with TypeScript. Many projects resort to Babel for various reasons, which is not able to compile const enums ([ref](https://babeljs.io/docs/en/babel-plugin-transform-typescript#caveats)). Luckily, `klighd-core` uses the TypeScript compiler and is able to compile const enums. :smiley:

Reply:
I think i remembered (this)[https://github.com/kieler/klighd-vscode/pull/1/files#diff-ecdc0f4b7f76b34d9bf26572c9b2134978bd14aca793765e2ae514a1a3057c03R78], but that is not an issue.
Resolving conversation.